sycl/test-e2e/Benchmarks/SYCLFrameworkOptimization/compare.py

```diff
@@ -0,0 +1,44 @@
+# -*- Python -*-
+"""Tool parses output of FloydWarshall programs and compares relatively
+the execution time both of them.
+"""
+
+import argparse
+
+
+def read_exec_result(filename):
+    with open(filename, 'r') as f:
+        for line in f:
+            if not line.startswith("Time: "):
+                continue
+            
+            line = line.strip("Time: ")
+            return float(line.split()[0])
+        
+        raise Exception('No found the following line: "Time: ", filename: {0}'.format(filename))
+
+
+parser = argparse.ArgumentParser(prog="compare.py")
+parser.add_argument("--files", action="store", dest="files",
+                    help="2 files that are expected to be outputs of FloydWarshall program runs",
+                    required=True)
+parser.add_argument("--diff", action="store", dest="diff",
+                    help="Relative diff used to compare two execution times",
+                    type=float, required=True)
+
+args = parser.parse_args()
+files = args.files
+files = files.split(',')
+if len(files) != 2:
+    raise Exception("Only 2 output files are expected")
+
+diff = args.diff
+if not (0. < diff < 1.):
+    raise "relative diff must be in (0, 1)"
+
+exec_results = [read_exec_result(f) for f in files]
+if exec_results[1] > diff * exec_results[0]:
```

Comment:
sorry for the late review. i have a higher level question, are we sure we want to add benchmarks as part of our e2e suite where it will fail if the performance is too poor. are we guaranteed the optimized case will always beat the `O0` case even wtih random os scheduler/hardware utilization/etc? i want to make sure all the failures of this test are actual failures we need to investigate and not just noise

Reply:
I agree that the bigger issue of adding benchmark tests needs to be discussed in a bigger forum (may be open an issue). For this scenario, we expect that the user will report any run where O1/O2/O3 compilations are slower than O0 compilation. So, it is advisable to catch this issue using a pre-existing test.

Thanks

Reply:
The test is picked up in a way that a perf diff between them is around 80%.
And for an observability this is the best place to put this test.

Reply:
I'm strongly against this.

Reply:
I think it should be a test under `sycl/test` that would verify LLVM IR/SPIRV at different stages of compilation instead and the source code of the test should really be tiny.

Reply:
@aelovikov-intel
The aim of this feature to make programs more efficient with -O0.
Did I get it right that you propose to check LLVM IR output?
If yes, how do you see it possible to check performance by looking at IR?

Reply:
Performance == optimizations enabled + optimizations do the right thing. This feature is about the former, and verifying LLVM IR is sufficient for its testing, IMO.

Reply:
`optimizations do the right thing`
And what exactly "the right thing" could be?

Reply:
Are you suggesting adding every possible sycl benchmark/application to those tests? If not, why is this one so special?

Reply:
The difference is that our team doesn't own these benchmarks and applications. However, we own this particular feature.

Reply:
But we also own all the other features of sycl that affect performance of every existing sycl benchmark/application. I still don't see why this one has to be treated specially.

+ @bader , @steffenlarsen , @sergey-semenov

Reply:
I agree with @aelovikov-intel. 
Compile-only check that `-fsycl-optimize-non-user-code` enables certain passes should be good enough for pre-commit tests.